Essentials/src/com/earth2me/essentials/commands/Commandbuy.java

```diff
@@ -0,0 +1,90 @@
+package com.earth2me.essentials.commands;
+
+import com.earth2me.essentials.Trade;
+import com.earth2me.essentials.User;
+import com.earth2me.essentials.craftbukkit.InventoryWorkaround;
+import com.earth2me.essentials.utils.NumberUtil;
+import com.google.common.collect.Lists;
+
+import org.apache.commons.lang.math.NumberUtils;
+import org.bukkit.Server;
+import org.bukkit.World;
+import org.bukkit.inventory.ItemStack;
+
+import static com.earth2me.essentials.I18n.tl;
+
+import java.math.BigDecimal;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.logging.Level;
+
+public class Commandbuy extends EssentialsCommand {
+    public Commandbuy() {
+        super("buy");
+    }
+
+    @Override
+    public void run(final Server server, final User user, final String commandLabel, final String[] args) throws Exception {
+        if (args.length < 2 || !NumberUtil.isInt(args[1])) {
```

Comment:
If NumberUtil.isInt fails here, it isn't really a case of not enough arguments - this will result in a confusing error message for the user.

Reply:
I have not written much Java and this is my first time contributing to this repo.  What function should I use to perform this check?  Essentially, I just want to make sure that args[1] can be parsed to an integer value and is not a string/float/etc.

Reply:
I took my best shot at this.  Let me know if my latest code works.